specs/protocol/derivation.md

```diff
@@ -363,29 +363,49 @@ where:
 - `is_last` is a single byte with a value of 1 if the frame is the last in the channel, 0 if there are frames in the
   channel. Any other value makes the frame invalid (it must be ignored by the rollup node).
 
-[batcher-spec]: batching.md
-
 ### Channel Format
 
-A channel is encoded as `channel_encoding`, defined as:
+[channel-format]: #channel-format
+
+A channel is encoded by applying a streaming compression algorithm to a list of batches:
 
 ```text
 rlp_batches = []
 for batch in batches:
     rlp_batches.append(batch)
-channel_encoding = compress(rlp_batches)
 ```
 
 where:
 
 - `batches` is the input, a sequence of batches byte-encoded as per the next section ("Batch Encoding")
 - `rlp_batches` is the concatenation of the RLP-encoded batches
-- `compress` is a function performing compression, using the ZLIB algorithm (as specified in [RFC-1950][rfc1950]) with
-  no dictionary
-- `channel_encoding` is the compressed version of `rlp_batches`
+
+Prior to the Fjord upgrade,
+
+```text
+channel_encoding = zlib_compress(rlp_batches)
+```
+
+where zlib_compress is the ZLIB algorithm (as specified in [RFC-1950][rfc1950]) with no dictionary.
 
 [rfc1950]: https://www.rfc-editor.org/rfc/rfc1950.html
 
+After the Fjord upgrade, there are two possible channel encoding formats. The legacy format,
+which is exactly as above, and a versioned format:
+
+```text
+channel_encoding = `version_byte ++ compress(rlp_batches)`
+```
+
+For the versioned format, the `version_byte` must never have its 4 lower order bits set to 0x1000,
```

Comment:
I think using `0x...` here is misleading, because it's not in hex, but in binary notation. So maybe better
```suggestion
For the versioned format, the `version_byte` must never have its 4 lower order bits set to `0b1000 = 8`,
```

We should also elaborate on why this selects for zlib and maybe reference Page 5 from the zlib rfc that explains that this is set to 8. Maybe we should also forbid 15 since it's reserved in the zlib spec? But maybe that's too paranoid.

Reply:
reworded, and added 15 to reserved bit list.